Learn15_Structures/Learn15_Structures/Learn15_Structures.cpp

```diff
@@ -0,0 +1,145 @@
+﻿#include <iostream>
+#include <string>
+#include <algorithm>
+
+const unsigned Number_Students = 4;
+const unsigned Number_Grades = 4;
+
+struct Student
+{
+    std::string name;
+    unsigned marks[Number_Grades];
+};
+
+
+//Task 1
+double calculateAverage(const Student& student, unsigned numberGrades)
+{
+    double sum = 0;
+    for (int i = 0; i < numberGrades; i++)
+    {
+        sum += student.marks[i];
+    }
+
+    return sum / numberGrades;
+}
+
+void printAverage(Student* students, int size)
+{
+    for (int i = 0; i < size; i++)
+    {
+        std::cout << "Student: " << students[i].name << "\t\tAverage marks: " << calculateAverage(students[i], Number_Grades) << std::endl;
+    }
+}
+
+//Task 2
+bool compare(const Student& student1, const Student& student2)
+{
+    return calculateAverage(student1, Number_Grades) > calculateAverage(student2, Number_Grades);
+}
+
+void sort(Student* students, unsigned size)
+{
+    std::sort(students, students + size, compare);
+}
+
+//Task 3
+Student* getTopStudent(Student* students, unsigned size)
+{
+    if (size == 0) return nullptr;
+    
+    //Method 1
+    sort(students, Number_Students);
+    Student* topStudent = &students[0];
+
+    //Method 2
+    /*Student* topStudent = &students[0];
+    double highestAverage = calculateAverage(students[0], Number_Grades);
+
+    for (unsigned i = 1; i < size; ++i)
+    {
+        double currentAverage = calculateAverage(students[i], Number_Grades);
+        if (currentAverage > highestAverage)
+        {
+            highestAverage = currentAverage;
+            topStudent = &students[i];
+        }
+    }*/
```

Comment:
Is the only correct way) a linear search is much better than sorting in a first place